include/s3select_functions.h

```diff
@@ -846,6 +855,119 @@ struct _fn_version : public base_function
   }
 };
 
+struct _fn_isnull : public base_function
+{
+
+  value res;
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter = args->begin();
+    base_statement* expr = *iter;
+    value expr_val = expr->eval();
+    if ( expr_val.is_null()) {
+      result->set_value((int64_t)true);
+    } else {
+      result->set_value((int64_t)false);
+    }
+    return true;
+  }
+};
+
+struct _fn_in : public base_function
+{
+
+  value res;
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter = args->begin();
+    bs_stmt_vec_t::iterator it = args->end();
+    base_statement* main_expr = *it;
+    value main_expr_val = main_expr->eval();
+    int args_size = args->size();
+    while (args_size > 1)
+    {
+      base_statement* expr = *iter;
+      value expr_val = expr->eval();
+      iter++;
+      if ( expr_val.type == main_expr_val.type) {
+        if ( expr_val == main_expr_val) {
+          result->set_value((int64_t)true);
+          return true;
+        } 
+    } 
+      args_size--;
+    }
+    result->set_value((int64_t)false);
+    return true;
+  }
+};
+
+struct _fn_like : public base_function
+{
+
+  value res;
+
+  std::string transform(std::string s)
+  {
+    bool startswith = (s[0] == '%' && s[s.size()-1] != '%');
+    bool endswith = (s[0] != '%' && s[s.size()-1] == '%');
+    bool startend = (s[0] == '%' && s[s.size()-1] == '%');
+    if (startswith) {
+      if (s[1] == '%') s.erase(s.begin()+1);
+      std::replace(s.begin(), s.begin()+1, '%', '^');
+      s.insert(1, ".");
+      s.insert(2, "*");
+      s.append("$");
+    } else if (endswith) {
+      s.insert(0, "^");
+      std::replace(s.end()-1, s.end(), '%', '$');
+      s.insert(s.size()-1,".");
+      s.insert(s.size()-1,"*");
+    } else if (startend) {
+      std::replace(s.begin(), s.begin()+1, '%', '^');
+      s.insert(1, ".");
+      s.insert(2, "*");
+      std::replace(s.end()-1, s.end(), '%', '$');
+      s.insert(s.size()-1,".");
+      s.insert(s.size()-1,"*");
+    } else {
+      std::string findThis = "%";
+      std::string replaceWith = ".*";
+      s.insert(0,"^");
+      s.append("$");
+      std::size_t pos = 0;
+      while ((pos = s.find(findThis, pos)) != std::string::npos)
+      {
+        s.replace(pos, findThis.length(), replaceWith);
+        pos += replaceWith.length();
+      }
+      std::replace(s.begin(), s.end(), '_', '.');
+    }
+    return s;
+  }
+  
+  bool operator()(bs_stmt_vec_t* args, variable* result)
+  {
+    bs_stmt_vec_t::iterator iter = args->begin();
+    base_statement* expr = *iter;
+    iter++;
+    base_statement* main_expr = *iter;
+    value expr_val = expr->eval();
+    value main_expr_val = main_expr->eval();
+    std::string a = expr_val.to_string();
+    std::string s = transform(a);
+    std::regex x{s};
```

Comment:
the regex is actually compiled expression, which _regex_match_ is using.
/transform/ and /regex/ should called once per query ,from performance perspective it's a big difference.
thus, the optimized way is to set /x/ once on construction (should add construction).

/push_like_predicate/ is the place to call constructor with the like expression.
the parser definition should modified also , to accept constant string and not to accept general expression.